api/src/utils/helpers/flatten.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2025 Hexastack. All rights reserved.
+ *
+ * Licensed under the GNU Affero General Public License v3.0 (AGPLv3) with the following additional terms:
+ * 1. The name "Hexabot" is a trademark of Hexastack. You may not use this name in derivative works without express written permission.
+ * 2. All derivative works must include clear attribution to the original creator and software, Hexastack and Hexabot, in a prominent location (e.g., in the software's "About" section, documentation, and README file).
+ */
+
+/**
+ * Flattens a nested object into a single-level object with dot-separated keys.
+ * @param data - The data object to flatten
+ * @param prefix - The optional base key to prefix to the current object's keys
+ * @param result - The optional accumulator for the flattened object  data
+ * @returns A new object with flattened keys
+ */
+export const flatten = (
```

Comment:
Two distinct paths can collapse into the same dotted key if the original keys themselves contain dots ({ 'a.b': { c: 1 } }). Can you check ?

Reply:
Yes, i will check this use case :technologist:

Reply:
The logic will return always the latest occurrence as final value 
`flatten({ 'a.b.c': 1, 'a.b': { c: 2 }, a: { b: { c: 3 } } });`
will return `{a.b.c: 3}` in this case